Altimeter/Src/altimeter.cpp

```diff
@@ -0,0 +1,253 @@
+#include <altimeter.hpp>
+
+
+MS5611::MS5611(SPI_HandleTypeDef *spi_handle, GPIO_TypeDef *cs_port, GPIO_TypeDef *ps_port, uint16_t cs_pin, uint16_t ps_pin){
+	spi_handle_ = spi_handle;
+	cs_port_ = cs_port;
+	ps_port_ = ps_port;
+	cs_pin_ = cs_pin;
+	ps_pin_ = ps_pin;
+	HAL_GPIO_WritePin(ps_port_, ps_pin_, GPIO_PIN_RESET);
+
+	communication_success_ = true;
+
+	reset();
+	getPromData();
+
+	// Grab the sensor's altitude above sea level while on some surface and assign it to base_elev_
+	// Subtract this from measurements at greater heights for the height above that initial surface.
+	float base_elevation = 0.0f;
+	if (communication_success_) {   //if HAL_SPI_TransmitReceive_IT doesn't fail in reset or getPromData()
+
+		float altitude_above_sea_level_measurement = 0;
+		int num_successful_measurements = 0;
+		for (int i = 0; i < 100; i++) { 			//take a few measurements to make sure they are accurate/consistent.
+			communication_success_ = true;
+			communication_success_ = getAltitudeAboveSeaLevel(altitude_above_sea_level_measurement);
+			osDelay(10);
+			if (communication_success_) {			//count the number of successful measurements.
+				base_elevation = altitude_above_sea_level_measurement;
+				num_successful_measurements += 1;
+			}else{
+				communication_success_ = false;
+			}
+		}
+	}
+
+	base_elev_ = base_elevation;
+}
+
+void MS5611::reset(){
+	HAL_GPIO_WritePin(cs_port_, cs_pin_, GPIO_PIN_SET);
+	HAL_GPIO_WritePin(cs_port_, cs_pin_, GPIO_PIN_RESET);
+
+	uint8_t tx_data[3] = {RESET_COMMAND, 0xFF, 0xFF};
+	uint8_t rx_data[3];
+	uint16_t data_size = 3;
+
+	spi_status_ = HAL_SPI_TransmitReceive_IT(spi_handle_, tx_data, rx_data, data_size);
+	if (spi_status_ != HAL_OK) {
```

Comment:
It seems to me the `spi_status_` in this class could totally be initialized locally in every function instead of making it a private member variable.